numpy/core/arrayprint.py

```diff
@@ -521,7 +513,10 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
-    if a.size == 0:
+    if a.shape == () and a.dtype.names is None:
```

Comment:
My suggestion would be for other uses of `array2string`, i.e., to allow people to say that they just want the same style for arrays and scalars (it could also be assigning the same `dict` to `style` as for `formatter`, but I think this is less clean). This is handy for astropy at least, although it would have been even better if one could also set `style` with `set_printoptions`.

Indeed, it is somewhat surprising that one cannot:
```
In [2]: np.set_printoptions(formatter={'all': lambda x: 'haha'})

In [3]: np.array([1, 2])
Out[3]: array([haha, haha])

In [4]: np.array(1)
Out[4]: array(1)
```